icevision/backbones/resnest_fpn.py

```diff
@@ -0,0 +1,120 @@
+__all__ = [
+    "resnest50",
+    "resnest101",
+    "resnest200",
+    "resnest269",
+]
+
+from icevision.imports import *
+from icevision.utils import *
+from icevision.backbones.resnet_fpn import patch_param_groups
+from torchvision.ops.feature_pyramid_network import LastLevelMaxPool
+from torchvision.models.detection.backbone_utils import (
+    resnet_fpn_backbone,
+    BackboneWithFPN,
+)
+import resnest.torch
+
+
+# ResNeSt Backbones
+class Identity(nn.Module):
+    def __init__(self):
+        super(Identity, self).__init__()
+
+    def forward(self, x):
+        return x
+
+
+def resnest_fpn_backbone(
+    resnest_name,
+    pretrained,
+    # norm_layer=misc_nn_ops.FrozenBatchNorm2d,
+    trainable_layers=3,
+    returned_layers=None,
+    extra_blocks=None,
+):
+    """
+    Constructs a specified ResNet backbone with FPN on top. Freezes the specified number of layers in the backbone.
+
+    Examples::
+
+        >>> from torchvision.models.detection.backbone_utils import resnet_fpn_backbone
+        >>> backbone = resnet_fpn_backbone('resnet50', pretrained=True, trainable_layers=3)
+        >>> # get some dummy image
+        >>> x = torch.rand(1,3,64,64)
+        >>> # compute the output
+        >>> output = backbone(x)
+        >>> print([(k, v.shape) for k, v in output.items()])
+        >>> # returns
+        >>>   [('0', torch.Size([1, 256, 16, 16])),
+        >>>    ('1', torch.Size([1, 256, 8, 8])),
+        >>>    ('2', torch.Size([1, 256, 4, 4])),
+        >>>    ('3', torch.Size([1, 256, 2, 2])),
+        >>>    ('pool', torch.Size([1, 256, 1, 1]))]
+
+    Arguments:
+        backbone_name (string): resnet architecture. Possible values are 'ResNet', 'resnet18', 'resnet34', 'resnet50',
+             'resnet101', 'resnet152', 'resnext50_32x4d', 'resnext101_32x8d', 'wide_resnet50_2', 'wide_resnet101_2'
+        norm_layer (torchvision.ops): it is recommended to use the default value. For details visit:
+            (https://github.com/facebookresearch/maskrcnn-benchmark/issues/267)
+        pretrained (bool): If True, returns a model with backbone pre-trained on Imagenet
+        trainable_layers (int): number of trainable (not frozen) resnet layers starting from final block.
+            Valid values are between 0 and 5, with 5 meaning all backbone layers are trainable.
+    """
+    backbone = resnest_name(pretrained=pretrained)
```

Comment:
I guess this is the only line different from the pytorch implementation? The problem was that they were using `module.__dict__` to get the model right?

Reply:
Yes. The other small difference is that we are not using this argument `norm_layer=misc_nn_ops.FrozenBatchNorm2d`. We know that choice is debatable given the fact that we usually train the BatchNorm layer. In the case of the `resnet_fpn` they pass it to the resnet method like this:
```
backbone = resnet.__dict__[backbone_name](
        pretrained=pretrained,
        norm_layer=norm_layer)
```